### PR TITLE
[Group] Fix Group Write Attribute sending and receiving path

### LINKED ISSUE #11604: Update attribute path for groupcast messages
#### Problem
Attribute path still expect endpointId and nodeId. Both of which are not available in a groupcast context.

 #### Proposed Solution
Fix the attribute path when https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/4618/commits/d073ebb8b111c0803008d933dbe57dd695f490d0 land

### LINKED ISSUE #11980: Verify Group Msg reception
#### Problem
While Group Msg reception is supposed to work, it wasn't really tested other than with some Unit Test.

 #### Proposed Solution
Make sure that upon reception of a Group Msg packet (groupId, GroupSession present, etc.) it is routed through the correct path inside the Matter SDK and that no unwanted behaviour is triggered (e.g. sending an Ack).

We can test/achieve this with the `TestGroupMessaging` yaml test case

### PR DESCRIPTION
**Pull request by jepenven-silabs**
#### Problem
Fix #11980 
Fix #11604 

#### Change overview
Enable the complete test case (Group Attribute Write + Unicast Read)

#### Testing
With the enabled yaml test case

### CODE REVIEW

File: src/app/WriteHandler.cpp [hunk 1]
```diff
@@ -131,9 +138,20 @@ CHIP_ERROR WriteHandler::ProcessAttributeDataIBs(TLV::TLVReader & aAttributeData
...
+        if (mpExchangeCtx->IsGroupExchangeContext())
+        {
+            // TODO retrieve Endpoint ID with GroupDataProvider using GroupId and FabricId
+            // Issue 11075
 
-        err = attributePath.GetEndpoint(&(clusterInfo.mEndpointId));
-        SuccessOrExit(err);
+            // Using endpoint 0 for test purposes
+            clusterInfo.mEndpointId = 0;
+            SuccessOrExit(err);
```
Comment (tcarmelveilleux) [inline src/app/WriteHandler.cpp:148]: This is superfluous

File: src/transport/SessionManager.cpp [hunk 2]
```diff
@@ -131,6 +131,8 @@ CHIP_ERROR SessionManager::PrepareMessage(SessionHandle sessionHandle, PayloadHe
             {
                 return CHIP_ERROR_INTERNAL;
             }
+            // TODO #11911
```
Comment (tcarmelveilleux) [inline src/transport/SessionManager.cpp:134]: Could you add a quick description here?

**Review by tcarmelveilleux — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]
Approving assuming my comments are addressed....

File: src/app/WriteHandler.cpp (hunk 1 above)
Comment (jepenven-silabs) [reply src/app/WriteHandler.cpp:148]: Removed

File: src/transport/SessionManager.cpp (hunk 2 above)
Comment (jepenven-silabs) [reply src/transport/SessionManager.cpp:134]: Done

File: src/app/WriteClient.cpp [hunk 3]
```diff
@@ -244,6 +244,12 @@ CHIP_ERROR WriteClient::SendWriteRequest(SessionHandle session, System::Clock::T
     mpExchangeCtx = mpExchangeMgr->NewContext(session, this);
     VerifyOrExit(mpExchangeCtx != nullptr, err = CHIP_ERROR_NO_MEMORY);
 
+    if ((mpExchangeCtx->IsGroupExchangeContext() && !mpExchangeCtx->IsInitiator()))
+    {
+        err = CHIP_ERROR_INTERNAL;
+        SuccessOrExit(err);
+    }
+
```
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/app/WriteClient.cpp:252]: Yeah, this is some debugging code that was left in the PR.  Removing

File: src/messaging/ExchangeContext.cpp [hunk 4]
```diff
@@ -445,8 +451,11 @@ CHIP_ERROR ExchangeContext::HandleMessage(uint32_t messageCounter, const Payload
         MessageHandled();
     });
 
-    ReturnErrorOnFailure(
-        mDispatch->OnMessageReceived(messageCounter, payloadHeader, peerAddress, msgFlags, GetReliableMessageContext()));
+    if (!IsGroupExchangeContext())
```
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/messaging/ExchangeContext.cpp:454]: Good catch. I wasn't a fan of bypassing the `mDispatch` but all that was done inside seemed to be strictly related to Unicast messages. Opening an Issue to fix this #12101

**Review by jmartinez-silabs — APPROVED** [review]

**Review by yufengwangca — APPROVED** [review]

**Pull request merged**